### ISSUE TITLE
New rule proposal: Multi-line statements must be surrounded by braces

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#717]
This rule is an intentional subset of the behavior of SA1503 (CurlyBracketsMustNotBeOmitted) targeting the behavior used by me in my own projects and now adopted by CoreFX.

**Category:** Layout
**Rule:** Multi-line statements must be surrounded by braces (improvements to this wording welcome)
**Description:** If the child of a compound statement does not fit on a single line, then curly braces must be used for the compound statement.

**Example:**

The following code does not trigger this rule:

```csharp
if (x == null)
    throw new ArgumentNullException(nameof(x), "some message");
```

The following code **does** trigger this rule:

```csharp
// The body of the `if` statement omits braces *and* spans multiple lines, thus
// triggering this diagnostic.
if (x == null)
    throw new ArgumentNullException(
        nameof(x),
        "some message);
```

To fix a violation of this rule, either insert curly braces or place the entire statement on a single line. The automatic code fix for this violation would always choose to insert braces.

**Comment by AArnott** [comment]
:+1:

**Comment by vweijsters** [comment]
:+1:

**Comment by Przemyslaw-W** [comment]
+1

2015-05-20 17:32 GMT+02:00 Vincent Weijsters <notifications@github.com>:
